src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,590 @@
+    #  - could try both
+    #  - is rels_indirect_source_mondo useful? used anywhere?
+    #  - add docs for each variable here like in prev section
+    # rels_indirect_mondo_source = ''  # not needed?
+    # todo: rels_indirect_source_and_1or2_ids_not_in_mondo unused. discard?
+    rels_indirect_source_mondo, rels_indirect_source_and_1or2_ids_not_in_mondo = _convert_edge_namespace(
+        rels_indirect_source, source_mondo_map)
+
+    # Determine hierarchy diferences -----------------------------------------------------------------------------------
+    # todo: was there a reason why the _direct cases were done using edges w/ source IDs and _indirect Mondo IDs?
+    # - find which edges appear in both Mondo and source, or only in one or the other
+    #   - direct
+    in_both_direct_source_edges: Set[RELATIONSHIP] = rels_direct_source.intersection(rels_direct_mondo_source)
+    in_source_only_direct_source_edges: Set[RELATIONSHIP] = rels_direct_source.difference(rels_direct_mondo_source)
+    in_mondo_only_direct_source_edges: Set[RELATIONSHIP] = rels_direct_mondo_source.difference(rels_direct_source)
+    #  - indirect
+    in_both_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_source_mondo.intersection(
+        rels_indirect_mondo_mondo)
+    in_source_only_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_source_mondo.difference(
+        rels_indirect_mondo_mondo)
+    in_mondo_only_indirect_mondo_edges: Set[RELATIONSHIP] = rels_indirect_mondo_mondo.difference(
+        rels_indirect_source_mondo)
+    # - add metadata to edges (labels, both source/Mondo IDs)
+    #  - direct
+    in_both_direct: List[Dict] = _edges_with_metadata_from_plain_edges(in_both_direct_source_edges, source_data_map)
+    in_mondo_only_direct: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_mondo_only_direct_source_edges, source_data_map)
+    in_source_only_direct: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_source_only_direct_source_edges, source_data_map)
+    #   - indirect
+    # TODO 1.1: indirect rels: which are useful? @case-3-and-5-finished
+    #  - drop the ones that aren't, then also drop their Set[RELATIONSHIP] equivalent vars
+    in_both_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(in_both_indirect_mondo_edges, mondo_data_map)
+    in_mondo_only_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_mondo_only_indirect_mondo_edges, mondo_data_map)
+    in_source_only_indirect: List[Dict] = _edges_with_metadata_from_plain_edges(
+        in_source_only_indirect_mondo_edges, mondo_data_map)
+
+    print()
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    # Case 2: SCR is direct in source, but indirect Mondo
+    pass  # no output for this case
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    # todo: indirect+direct should == ancestors. Why not?
```

Comment:
## indirect+direct != ancestors (@joeflack4 )
<details><summary>Details</summary>
<p>

In one step, I'm getting list of direct SCRs. In another step, I'm getting ancestors, and then extracting just the SCRs from that which are indirect.
I did a sanity check, expecting the set of indirect + direct SCRs to be the same as ancestors SCRs (they should be), but they are not.

I should double check that all 3 of these sets are using terms in the same namespace. Maybe that's the issue. If not, then I'm baffled and need to look further into this.

</p>
</details>

Reply:
@matentzn Took me a little while to find this issue. Something was wrong with my cache of `ancestors()`. Weird because I don't believe I've changed the code for that function, nor the `mondo.db` since I last cached that. Anyway, fixed.